### ISSUE TITLE
how to install / use

### ISSUE THREAD

**Issue by ghost** [iampersistent/ImgSeek-php#1]
Hi,
i cannot figure out how to install or how to use the class? thanks

**Comment by iampersistent** [comment]
Its been a while since I touched the code and its a bit incomplete. 

I've updated the readme with install instructions.

Basically there are two gateways, one is to a persistence layer, like a db and one for imgseek, which uses the xml-rpc client. The manager handles the interactions between the two layers and helps facilitate searching for matching images.

I didn't get around to implementing a persistence layer, but you can see a partial example in the tests. I implemented a simple memory persistence layer, which might give you a bit of a hint on what to do.

Sorry there isn't more there.

**Comment by ruudbwai** [comment]
Hi there, looks good, however, can't actually find the php equivalent of the imgseek engine, I expected to see some sort of image processing, i.e scale the image to some dimension (128x128), some sort of hashing (i noticed a c++ Haar tensorial 2d transform in the imgseek source), then some sort of difference check...but couldn't find anything of the sort in the classes supplied....do we have such a thing as ---
1 - hash image 1
2 - hash image 2
3 - compare difference, if difference is less than x , images as somewhat similar? (im aware of pHash, but imgseek's algorithms seem a lot better for what I'm trying to accomplish)
using the imgseek algorithms, that seem pretty spot on, just can't turn that into PHP :(
imgseek seems to know about rgb of an image, and maps these for comparison, which is a whole load more useful than pHash, which seems pretty blind to colour....

phew, lots said...

tl:dr, php equivalent of the imgseek image hashing/ difference calculator engine that's in c++

Any clues/help? 

Many thanks :D

**Comment by iampersistent** [comment]
Not really. ImgSeek has been dormant for a couple of years now and I think that a lot of the attention has shifted to using deep learning to deal with image recognition. 

This might help some https://www.youtube.com/channel/UCWN3xxRkmTPmbKwht9FuE5A/search?query=image The guy is a little over the top in his style, but if you hang with him, there's a quite a bit that can be learned.

**Comment by ruudbwai** [comment]
That was excellent, and yeah he’s a bit OTT....very interesting stuff about machine learning, so I’ll give that a go! Again thanks for pointing me in that direction 😁👍🏾
